indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/ObjectIdentifier.kt

```diff
@@ -88,33 +180,37 @@ class ObjectIdentifier @Throws(Asn1Exception::class) constructor(@Transient vara
          * @throws Asn1Exception all sorts of errors on invalid input
          */
         @Throws(Asn1Exception::class)
-        fun parse(rawValue: ByteArray): ObjectIdentifier = runRethrowing {
-            if (rawValue.isEmpty()) throw Asn1Exception("Empty OIDs are not supported")
-            val (first, second) =
-                if (rawValue[0] >= 80) {
-                    2u to rawValue[0].toUByte() - 80u
-                } else {
-                    rawValue[0].toUByte() / 40u to rawValue[0].toUByte() % 40u
-                }
-
-            var index = 1
-            val collected = mutableListOf(first, second)
-            while (index < rawValue.size) {
-                if (rawValue[index] >= 0) {
-                    collected += rawValue[index].toUInt()
-                    index++
-                } else {
-                    val currentNode = mutableListOf<Byte>()
-                    while (rawValue[index] < 0) {
-                        currentNode += rawValue[index] //+= parsed
-                        index++
-                    }
-                    currentNode += rawValue[index]
-                    index++
-                    collected += currentNode.decodeAsn1VarUInt().first
-                }
+        fun parse(rawValue: ByteArray): ObjectIdentifier = ObjectIdentifier(rawValue)
```

Comment:
is there a reason why this is a static member while the others are secondary constructors? seems weird

Reply:
usually, when you construct OIDs you are not handling bytes, you are either passing strings or a list of nodes. Here you are parsing (more like validating), encoded bytes. this is in line with the rest of the API

Reply:
reconsider this in light of the latest changes, where there is no parsing/validating involved?

Reply:
see my comment below. the first node requires validation